Fix #2 by updating SDK version, build tools, etc

mitchhentges opened the pull request.

pvagner commented:
It builds fine for me after applying it.

ara4n commented on gradlew:
```diff
@@ -42,11 +42,6 @@ case "`uname`" in
-if $cygwin ; then
-    [ -n "$JAVA_HOME" ] && JAVA_HOME=`cygpath --unix "$JAVA_HOME"`
-fi
-
```
why is cygwin being removed here?

ara4n replied:
oh, i see, it's been shifted into the JAVACMD below.

mitchhentges replied:
[Generating the gradle wrapper](https://docs.gradle.org/current/userguide/gradle_wrapper.html) automatically generates `./gradlew` and `gradle.bat`, I'm not sure what the changes do, but I know it's more up-to-date and Gradle 2.11

ara4n commented on console/src/main/java/org/matrix/console/services/EventStreamService.java:
```diff
@@ -641,10 +635,15 @@ private Notification buildNotification() {
-        notification.setLatestEventInfo(this, getString(R.string.app_name),
-                "Listening for events",
-                pi);
-        notification.flags |= Notification.FLAG_NO_CLEAR;
```
how come we're removing FLAG_NO_CLEAR?

pvagner replied:
In order to build with SDK 23, I had to use [Notification.Builder](https://developer.android.com/reference/android/app/Notification.Builder.html) rather than using [Notification](https://developer.android.com/reference/android/app/Notification.html) directly.
Notification.Builder has an equivalent method [setOngoing](https://developer.android.com/reference/android/app/Notification.Builder.html#setOngoing%28boolean%29)(boolean ongoing) doing the same thing I'm calling one line below.
Hopefully I haven't interpreted it wrong.
Thanks for the review anyway.

ara4n commented:
i've added various questions, plus I suspect we'll be quite paranoid about upgrading to 23 and will test it a bunch before merging.

ylecollen commented:
1 - i tested gradle-2.11 this week-end and it crashed a few times whereas it does not crash with oldest gradle versions.

2 - the build tools and the sdk version should be the same i.e. SDK 23 and builtools 23 (+ compat).
It is not common to use an old SDK with a recent buildtools (because of compat issues).

mitchhentges commented:
1: Remaining with an old version of software isn't a good idea, because eventually it will be unsupported. If we keep up with the current version, not only will we gain a bunch of performance improvements and bug fixes, but we'll be more agile on future changes.

I'm surprised that you've had issues with `gradle` 2.11 crashing. Google has developers use gradle directly, and they're using `gradle` 2.10/2.11 in their next build of Android Studio (the one with continuous deployment). If Google is recommending it, it should be a safe bet.

2: I've updated the builds tools and SDK version! I had to change notifications a little bit, but it still works (at least on an Android M device). I've changed the target to SDK 23, because the only _real_ change is that permissions are requested on-the-fly, and I think that we already deal with that

mitchhentges commented on console/build.gradle:
```diff
@@ -87,11 +87,11 @@ def gitRevisionDate() {
+    compile 'com.android.support:appcompat-v7:23.1.1'
+    compile 'com.android.support:cardview-v7:23.1.1'
+    compile 'com.android.support:recyclerview-v7:23.1.1'
     compile 'com.google.android.gms:play-services-gcm:8.3.0'
```
These google dependencies are out-of-date, but to upgrade them, I need to upgrade the `google-services` plugin. The next version of the `google-services` plugin is a beta, so it's probably safer to wait for its full release before these dependencies are upgraded

mitchhentges commented on console/src/main/java/org/matrix/console/services/EventStreamService.java:
```diff
@@ -640,12 +635,19 @@ private Notification buildNotification() {
-        notification.setLatestEventInfo(this, getString(R.string.app_name),
-                "Listening for events",
-                pi);
-        notification.flags |= Notification.FLAG_NO_CLEAR;
```
As @pvager noted on the comment on the old commit: `setOngoing(true)` replaces manually adding `FLAG_NO_CLEAR`.

mitchhentges commented on console/src/main/java/org/matrix/console/services/EventStreamService.java:
```diff
@@ -640,12 +635,19 @@ private Notification buildNotification() {
 
         PendingIntent pi = PendingIntent.getActivity(this, 0, i, 0);
-
-        notification.setLatestEventInfo(this, getString(R.string.app_name),
```
`setLatestEventInfo` was removed, so I needed to build a `Notification` using the `NotificationCompat.Builder` mechanism
